TESTS/netsocket/udp/udpsocket_echotest.cpp

```diff
@@ -85,7 +88,17 @@ void UDPSOCKET_ECHOTEST()
                 printf("[Round#%02d - Sender] error, returned %d\n", s_idx, sent);
                 continue;
             }
-            recvd = sock.recvfrom(NULL, rx_buffer, pkt_s);
+
+            do {
+                received_duplicate_packet = false;
+                recvd = sock.recvfrom(NULL, rx_buffer, pkt_s);
+                //Check if received duplicated packet
+                if (s_idx > 0 && recvd == pkt_sizes[s_idx - 1] && memcmp(rx_last, rx_buffer, recvd) == 0) {
+                    printf("[Round#%02d - Receiver] info, received duplicate packet, returned %d\n", s_idx, recvd);
+                    received_duplicate_packet = true;
+                }
+            } while (received_duplicate_packet);
+
```

Comment:
I don't like this logic.

You assume that all duplicates will be received in order.

This might be true for ECHOTEST, but might not be true for ECHOTEST_BURST.

So while re-working this, can you change logic so that it tolerates re-ordering as well.

I would assume this kind of logic might work:
```
// List of packet sizes to send
static const int pkt_sizes[PKTS] = {...);
// List of boolean flags to mark each size as "received"
static bool pkt_received[PKTS] = {false};

// ... in the test case
for (int retry_cnt = 0; retry_cnt <= 2; retry_cnt++) {
    ...
    sent = sock.sendto(udp_addr, tx_buffer, pkt_s);
    if (...) {
        // errors..
    }
    do {
        recvd = sock.recvfrom(...);
        // Check that packet is what we sent
        if (recvd == pkt_sizes[s_idx] && memcmp(rx_last, rx_buffer, recvd) == 0)
            pkt_received[s_idx] = true;
        }
    } while(pkt_received[s_idx] == false);
}
```

So just accept what we were waiting for, ignore the rest.

Reply:
I've changed logic. Now responses for all sends and duplicates of this responses are accepted. Other packets with incorrect size are interpreted as error.